packages/x-data-grid-premium/src/hooks/features/cellSelection/useGridCellSelection.ts

```diff
@@ -186,7 +186,7 @@ export const useGridCellSelection = (
           ...Object.entries(fields).reduce<{ id: GridRowId; field: string }[]>(
             (selectedFields, [field, isSelected]) => {
               if (isSelected) {
-                selectedFields.push({ id: idToIdLookup[id], field });
+                selectedFields.push({ id: apiRef.current.getRowId(rowsLookup[id]), field });
```

Comment:
```suggestion
                selectedFields.push({ id, field });
```
This should be enough, since `id` is coming from `currentVisibleRows.rows` and should be of correct type?

Reply:
Looks like it.
Was it there in the first place because we just miss it?

Reply:
I don't know.
We had some issues in the past, using a stringified row id, but it was a number, so we couldn't match the row properly. I think this was the purpose for the lookup you're removing.
But in this specific case it looks like we don't need that.